src/store/helpers.ts

```diff
@@ -27,8 +27,8 @@ export const getThumb = (thumbnails: Thumbnail[], path: string, large = true) =>
         if (thumb.relative_path && thumb.relative_path.length > 0) {
           const url = new URL(apiUrl ?? document.location.origin)
           url.pathname = (path === '')
-            ? `/server/files/gcodes/${thumb.relative_path}`
-            : `/server/files/gcodes/${path}/${thumb.relative_path}`
+            ? '/server/files/gcodes/' + encodeURIComponent(thumb.relative_path)
+            : '/server/files/gcodes/' + encodeURIComponent(path) + '/' + encodeURIComponent(thumb.relative_path)
```

Comment:
I haven't tested this yet but the fix makes sense.

Having said that, it would be preferable to keep using string interpolation, like this:

```suggestion
            ? `/server/files/gcodes/${encodeURIComponent(thumb.relative_path)}`
            : `/server/files/gcodes/${encodeURI(path)}/${encodeURI(thumb.relative_path)}`
```

Please note that I also replaced `encodeURIComponent` with `encodeURI` as I believe the former would encode characters like "/" when we need to keep those as is ([context](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI#encodeuri_vs_encodeuricomponent)).

Reply:
yes, ok. Please note that you missed one replacement on line 30.

Reply:
Good eye @bhelm, I've just updated the suggestion!